src/main/java/htsjdk/variant/variantcontext/writer/VCFWriter.java

```diff
@@ -228,4 +228,10 @@ public void add(final VariantContext context) {
             throw new RuntimeIOException("Unable to write the VCF object to " + getStreamName(), e);
         }
     }
+
+    @Override
+    public void setVcfHeader(VCFHeader header) {
+        this.mHeader = doNotWriteGenotypes ? new VCFHeader(header.getMetaDataInSortedOrder()) : header;
```

Comment:
Why are you changing the header on-the-fly like this when `doNotWriteGenotypes` is true? This at least requires an explanatory comment.